homeassistant/components/homekit/type_switches.py

```diff
@@ -241,13 +250,23 @@ def set_state(self, value: bool) -> None:
         _LOGGER.debug("%s: Set switch state to %s", self.entity_id, value)
         self.char_in_use.set_value(value)
         params = {ATTR_ENTITY_ID: self.entity_id}
-        service = SERVICE_TURN_ON if value else SERVICE_TURN_OFF
-        self.async_call_service(DOMAIN, service, params)
+
+        if self.domain == VALVE_DOMAIN:
+            service = SERVICE_OPEN_VALVE if value else SERVICE_CLOSE_VALVE
+        else:
+            service = SERVICE_TURN_ON if value else SERVICE_TURN_OFF
+
+        self.async_call_service(self.domain, service, params)
 
     @callback
     def async_update_state(self, new_state: State) -> None:
         """Update switch state after state changed."""
-        current_state = 1 if new_state.state == STATE_ON else 0
+
+        if self.domain == VALVE_DOMAIN:
+            current_state = 1 if new_state.state in [STATE_OPEN, STATE_OPENING, STATE_CLOSING] else 0
```

Comment:
Please make `[STATE_OPEN, STATE_OPENING, STATE_CLOSING]` a constant `set`

Reply:
Fixed in 61ddb28